packages/runtime/src/plugins/useUpstreamRetry.ts

```diff
@@ -0,0 +1,174 @@
+import { isOriginalGraphQLError } from '@envelop/core';
+import {
+  ExecutionRequest,
+  ExecutionResult,
+  isAsyncIterable,
+  mapMaybePromise,
+  MaybeAsyncIterable,
+  MaybePromise,
+} from '@graphql-tools/utils';
+import { DisposableSymbols } from '@whatwg-node/disposablestack';
+import { GatewayPlugin } from '../types';
+
+export interface UpstreamRetryOptions {
+  /**
+   * The maximum number of retries to attempt.
+   */
+  maxRetries: number;
+  /**
+   * The delay between retries in milliseconds.
+   * If the upstream returns `Retry-After` header, the delay will be the value of the header.
+   * @default 1000
+   */
+  retryDelay?: number;
```

Comment:
I didn't thought about it, but when it comes to retry policies, a backoff is often something important.

Like a simple exponential backoff, to avoid spamming the upstream if it is already having a hard time responding to requests :-P

Perhaps the retryDelay can be a function ? This way the user can define it's own backoff strategy.
Our default implementation could be to search for `Retry-After` header, and fallback to exponential backoff if not present ?

Reply:
That's what it does actually. It first look for the header if not present, it uses `retryDelay`

Reply:
Yes, but I mean having a `retryDelay` that is not a fixed value. `retryDelay` should increase after each try. Something like `retryDelay *= 2`

Reply:
Anyway what do you think now?